dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -120,9 +120,14 @@ public HostSelectionWithFallback(ConnectionPoolConfiguration config, ConnectionP
 			try { 
 				return hostPool.borrowConnection(duration, unit);
 			} catch (DynoConnectException e) {
-				lastEx = e;
-				cpMonitor.incOperationFailure(null, e);
-				useFallback = true;
+				if (e instanceof PoolExhaustedException) {
```

Comment:
Currently in this scenario the fall back does not occur immediately, it occurs after the timeout expires when polling the queue. By default this is 2 seconds, which is a long time to wait for a fall back. This will cause huge performance degradation (and did which is why the Hystrix timeout kicked in at the app layer).

For RF=3 the solution is good. Even in the case of a zone outage, the client will fail-fast and fall back to the other zones.

For the RF=2 case however, this is a good catch! The performance won't be degraded but the health tracker will infinitely be searching for a Dynomite node in the zone, which is not desirable. I will add the RF=2 client changes into this PR to account for this case.